pkg/pool-linear/contracts/yearn/YearnLinearPoolRebalancer.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "@balancer-labs/v2-interfaces/contracts/pool-linear/IYearnTokenVault.sol";
+import "@balancer-labs/v2-interfaces/contracts/pool-utils/ILastCreatedPoolFactory.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/math/Math.sol";
+
+import "../LinearPoolRebalancer.sol";
+
+contract YearnLinearPoolRebalancer is LinearPoolRebalancer {
+    using Math for uint256;
+
+    uint256 private immutable _divisor;
+
+    // These Rebalancers can only be deployed from a factory to work around a circular dependency: the Pool must know
+    // the address of the Rebalancer in order to register it, and the Rebalancer must know the address of the Pool
+    // during construction.
+    constructor(IVault vault, IBalancerQueries queries)
+        LinearPoolRebalancer(ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()), vault, queries)
+    {
+        IERC20 wrappedToken = ILinearPool(ILastCreatedPoolFactory(msg.sender).getLastCreatedPool()).getWrappedToken();
+
+        _divisor = 10**IYearnTokenVault(address(wrappedToken)).decimals();
+    }
+
+    function _wrapTokens(uint256 amount) internal override {
+        // Depositing from underlying (i.e. DAI, USDC, etc. instead of yvDAI or yvUSDC). Before we can
+        // deposit however, we need to approve the wrapper (yearn vault) in the underlying token.
+        _mainToken.approve(address(_wrappedToken), amount);
+        IYearnTokenVault(address(_wrappedToken)).deposit(amount, address(this));
```

Comment:
This line is failing in forked tests. Do you need to inform the recipient's address in the deposit function?

Reply:
Hey joão! We revised the PR to use the YearnShareValueHelper and added fork tests that pass.